src/goto-programs/restrict_function_pointers.cpp

```diff
@@ -244,9 +248,11 @@ function_pointer_restrictionst::merge_function_pointer_restrictions(
   return result;
 }
 
-function_pointer_restrictionst::restrictionst function_pointer_restrictionst::
-  parse_function_pointer_restrictions_from_command_line(
-    const std::list<std::string> &restriction_opts)
+function_pointer_restrictionst::restrictionst
+function_pointer_restrictionst::
+parse_function_pointer_restrictions_from_command_line(
+  const std::list<std::string> &restriction_opts,
+  const std::string &option_name)
```

Comment:
:no_entry_sign: This variable doesn't appear to be used

Reply:
I guess it should be used on line 274

Reply:
Though I think it would be better to have this throw an exception `invalid_restrictionst` or something, and have the callsite catch it and throw it as a command line exception, as this code doesn't need to know that it is coming from the command line or whatevre

Reply:
Good idea. I've reworked the error handling to now throw an `invalid_restriction_exceptiont`.